edb/common/signalctl.py

```diff
@@ -137,16 +143,21 @@ async def wait_for(self, fut, *, cancel_on=None):
             try:
                 try:
                     signal = await waiter
-                except asyncio.CancelledError as e:
+                except asyncio.CancelledError:
                     # Event 3: cancelled by outer code.
                     if not fut.done():
-                        fut.cancel(msg=e)
+                        fut.cancel()
+                        outer_cancelled_at_last = True
```

Comment:
The name is a bit cryptic. Maybe just call it `waiter_was_cancelled`?

Reply:
`waiter_was_cancelled` means "event 3 happened at least once", while we need "event 3 is the last event", because without ExceptionGroup, we can only raise one error, and it should be the last one in user-code. For examplye, if an outer cancel - say `asyncio.wait_for()` expired - was handled in a "finally" block in user-code, but we captured a sginal in the "finally" block, we should raise the SignalError instead of the CancelledError, in which case this flag should be `False` even though "waiter was cancelled" once.

On the same page I'm still open to suggestions for naming.